### PR TITLE
Return redirect if path begins with /node/

### PR DESCRIPTION
**Pull request by chrisrhymes**
Check if the path begins with `/node/` and return a DrupalRedirectRoute with the alias set as the destination. 

Additionally check that there is an alias set or that the alias is not the same as the path, otherwise it could cause a redirect loop. 

## To Test

- View a page in the CMS and make note of the url
- Press edit on a page in the CMS and then make note of the node id in the url
- Use the below graphql query and replace the 169 with the node id
- Ensure that the destination matches the url in the view page in the CMS
- Load the '/node/169' url (or whatever you have tested) in the frontend and it should be redirected to the destination

```graphql
{
  route(path: "/node/169") {
    __typename
    ... on DrupalRedirectRoute {
       __typename
      destination
      status
    }
  }
}
```

### CODE REVIEW

File: backend/src/modules/nc_graphql/src/Root.php [hunk 1]
```diff
@@ -82,6 +82,31 @@ public function getRoot() {
           $match_info = $this->router->match($args['path']);
           $entity = $this->findEntity($match_info);
           if ($entity && $entity instanceof Node) {
+
```
Comment (chrisrhymes) [inline backend/src/modules/nc_graphql/src/Root.php:85]: You can't seem to set the homepage alias to `/` in Drupal, it makes it blank. This if statement catches if the node type is homepage and then redirects to the root.

File: backend/src/modules/nc_graphql/src/Root.php [hunk 2]
```diff
@@ -82,6 +82,31 @@ public function getRoot() {
...
+            
+            // Redirect homepage to root
+            if ($entity->getType() === 'homepage') {
+              return $r + [
+                '__typename' => 'DrupalRedirectRoute',
+                'destination' => '/',
+                'status' => 301
+              ];
+            }
+
```
Comment (chrisrhymes) [inline backend/src/modules/nc_graphql/src/Root.php:94]: This if statement should catch all other `/node/id` paths and provide a redirect to the alias, unless there isn't an alias set.

**Review by DamienBurkeWN — CHANGES_REQUESTED** [review]
Works well for existing nodes. However, if you give it a node ID that doesn't exist (4 on my DB for instance), it bombs out hard and the frontend errors as a result rather than giving a 404

File: backend/src/modules/nc_graphql/src/Root.php [hunk 3]
```diff
@@ -96,7 +122,12 @@ public function getRoot() {
             '__typename' => 'DrupalAccessDeniedRoute',
             'reason' => 'TODO: Add reason for access denied to graphql response.',
           ];
+        } catch (Exception $e) {
```
Comment (chrisrhymes) [inline backend/src/modules/nc_graphql/src/Root.php:125]: I have added another catch for generic exception which returns a DrupalNotFoundRoute so the frontend should show a 404 in this case.

**Review by DamienBurkeWN — APPROVED** [review]
Perfick 👍

**Pull request merged**